internal/pkg/service/appsproxy/dns/dialer.go

```diff
@@ -0,0 +1,23 @@
+package dns
+
+import (
+	"net"
+	"time"
+)
+
+// DialTimeout specifies the default maximum connection initialization time.
+const DialTimeout = 2 * time.Second
+
+// KeepAlive specifies the default interval between keep-alive probes.
+const KeepAlive = 15 * time.Second
+
+func NewDialer() *net.Dialer {
+	return &net.Dialer{
+		Timeout:   DialTimeout,
+		KeepAlive: KeepAlive,
+		Resolver: &net.Resolver{
+			PreferGo:     true,
+			StrictErrors: true,
+		},
+	}
+}
```

Comment:
This dialer is used internally by the DNS client, to create connection to the DNS server, it sholdn't be public.
Also, the `NewClient` function may be private.

You need only a one public function:  `dns.NewDialer` / `dns.NewDialContextFn()` or somethind similar.

It can be confusing that there are 2 different dialers. One is used by the DNS client and another by the HTTP transport.